src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,742 @@
+          // for right-side drawer
+          dragX = -dragX;
+          touchX = containerWidth - touchX;
+          velocityX = -velocityX;
+        }
+
+        const gestureStartX = touchX - dragX;
+        let dragOffsetBasedOnStart = 0;
+
+        if (drawerType === DrawerType.FRONT) {
+          dragOffsetBasedOnStart =
+            gestureStartX > drawerWidth ? gestureStartX - drawerWidth : 0;
+        }
+
+        const startOffsetX =
+          dragX +
+          dragOffsetBasedOnStart +
+          (isDrawerOpen.value ? drawerWidth : 0);
+
+        const projOffsetX = startOffsetX + DRAG_TOSS * velocityX;
+
+        const shouldOpen = projOffsetX > drawerWidth / 2;
+
+        if (shouldOpen) {
+          animateDrawer(drawerWidth, velocityX);
+        } else {
+          animateDrawer(0, velocityX);
+        }
+      },
+      [
+        animateDrawer,
+        containerWidth,
+        drawerPosition,
+        drawerType,
+        drawerWidth,
+        isDrawerOpen.value,
+      ]
+    );
+
+    const openDrawer = (options: DrawerMovementOption = {}) => {
+      'worklet';
+      animateDrawer(
+        drawerWidth,
+        options.velocity ? options.velocity : 0,
+        options.speed
+      );
+    };
+
+    const closeDrawer = React.useCallback(
+      (options: DrawerMovementOption = {}) => {
+        'worklet';
+        animateDrawer(
+          0,
+          options.velocity ? options.velocity : 0,
+          options.speed
+        );
+      },
+      [animateDrawer]
+    );
+
+    const overlayDismissGesture = React.useMemo(
+      () =>
+        Gesture.Tap()
+          .maxDistance(25)
+          .onEnd(() => {
+            if (
+              isDrawerOpen.value &&
+              props.drawerLockMode !== DrawerLockMode.LOCKED_OPEN
+            ) {
+              closeDrawer();
+            }
+          }),
+      [closeDrawer, isDrawerOpen.value, props.drawerLockMode]
+    );
+
+    const overlayAnimatedStyle = useAnimatedStyle(() => ({
+      opacity: openValue.value,
+      backgroundColor: overlayColor,
+    }));
+
+    const renderOverlay = React.useCallback(() => {
+      return (
+        <GestureDetector gesture={overlayDismissGesture}>
+          <Animated.View
+            animatedProps={overlayAnimatedProps}
+            style={[styles.overlay, overlayAnimatedStyle]}
+          />
+        </GestureDetector>
+      );
+    }, [overlayAnimatedProps, overlayAnimatedStyle, overlayDismissGesture]);
+
+    const fillHitSlop = React.useMemo(
+      () => (isFromLeft ? { left: drawerWidth } : { right: drawerWidth }),
+      [drawerWidth, isFromLeft]
+    );
+
+    const panGesture = React.useMemo(() => {
+      return Gesture.Pan()
+        .activeCursor(activeCursor)
+        .mouseButton(mouseButton)
+        .hitSlop(drawerOpened ? fillHitSlop : edgeHitSlop)
+        .minDistance(drawerOpened ? 100 : 0)
+        .activeOffsetX(gestureOrientation * minSwipeDistance)
+        .failOffsetY([-15, 15])
```

Comment:
I think the only item that isn't already configurable here is `failOffsetY`, I could add a prop for configuring that, but i don't think there's any use case for doing so. 
`failOffsetY` is completely ignored when opening the drawer.
In all other cases, `failOffsetY` only really triggers when the swipe is almost completely vertical, and I'm actually not sure why that would be useful, since drawer closing is already cancelable by lifting the finger. 
Perhaps it could prevent some bugs with scroll list, but I'd have to test it further to check for that.
Maybe we should just remove it?

Also, aren't `minDistance` and `activeOffsetX` redundant? They both set the minimal distance a touch has to travel before activating the `Pan` gesture.

Reply:
I don't think it's necessary and we also shouldn't remove it. It's there so that vertical swipes don't block other gestures in any way (otherwise the pan could activate when swiping up-down).